tests/unit_tests/data_model/test_validate_data.py

```diff
@@ -771,3 +771,56 @@ def test_validate_data_dict_using_json_schema(caplog):
     with caplog.at_level(logging.DEBUG):
         data_validator._validate_data_dict_using_json_schema(valid_data, None)
     assert "Skipping validation of dict type" in caplog.text
+
+
+def test_get_value_and_units_as_lists():
+    data_validator = validate_data.DataValidator()
+
+    # Test with single value and unit
+    data_validator.data_dict = {"value": 100, "unit": "m"}
+    values, units = data_validator._get_value_and_units_as_lists()
+    assert values == [100]
+    assert units == ["m"]
+
+    # Test with list of values and units
+    data_validator.data_dict = {"value": [100, 200], "unit": ["m", "cm"]}
+    values, units = data_validator._get_value_and_units_as_lists()
+    assert values == [100, 200]
+    assert units == ["m", "cm"]
+
+    # Test with numpy array of values and units
+    data_validator.data_dict = {"value": np.array([100, 200]), "unit": np.array(["m", "cm"])}
+    values, units = data_validator._get_value_and_units_as_lists()
+    assert values == [100, 200]
+    assert units == ["m", "cm"]
+
+    # Test with unit as "null"
+    data_validator.data_dict = {"value": [100, 200], "unit": ["m", "null"]}
+    values, units = data_validator._get_value_and_units_as_lists()
+    assert values == [100, 200]
+    assert units == ["m", None]
+
+    # Test with mixed types
+    data_validator.data_dict = {"value": [100, 200], "unit": np.array(["m", "cm"])}
+    values, units = data_validator._get_value_and_units_as_lists()
+    assert values == [100, 200]
+    assert units == ["m", "cm"]
```

Comment:
I think this is already tested above in the second test (I assume copilot did this). You can change the first one to be all with the same unit or just delete this.

Reply:
Removed it. Indeed a copilot invention.